src/helics/core/SmallBuffer.hpp

```diff
@@ -279,15 +292,15 @@ class SmallBuffer {
             if (size > 0x010'0000'0000U) {
                 throw(std::bad_alloc());
             }
-            auto* ndata = new std::byte[size];
+            auto* ndata = new std::byte[size+8];
```

Comment:
Why the extra 8 bytes? Just a bit of extra capacity in case more data gets added?

Reply:
Now there is a fairly common use case with a need to add a single character, no sense in just adding a single character though. I suppose I could add some sort of exponential growth factor as well like std::string.